File: LeetCode/083 Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @param {ListNode} head
    # @return {ListNode}
    def deleteDuplicates(self, head):
        list_head = head
        while head != None and head.next != None:
            if head.val == head.next.val:
                head.next = head.next.next
            else:
                head = head.next
        return list_head

File: LeetCode/083 Remove Duplicates from Sorted List.cpp
class Solution {
 
 public:
  
  ListNode *deleteDuplicates(ListNode *head) {
    ListNode *first = head;
    ListNode *previous = head;
    ListNode *current = head;
    while (current && current->next) {
      current = current->next;
      if (current->val == previous->val) {
        previous->next = current;
      }
      previous = current;
    }
    return first;
  }

};
